astroquery/esa/hsa/tests/test_hsa_remote.py

```diff
@@ -3,168 +3,79 @@
 import tarfile
 
 import pytest
-from requests.exceptions import ChunkedEncodingError
 
 from ..core import HSAClass
 
-spire_chksum = [10233, 10762, 9019, 10869, 3944, 11328, 3921, 10999, 10959,
-                11342, 10974, 3944, 11335, 11323, 11078, 11321, 11089, 11314, 11108, 6281]
 
-pacs_chksum = [10208, 10755, 8917, 10028, 3924, 3935, 6291]
+pytestmark = pytest.mark.remote_data
 
 
-@pytest.mark.remote_data
-class TestHSARemote:
-    retries = 2
+PACS_ENDINGS = ["571.xml", "571.jpg", "214.fits.gz", "008.fits.gz",
+                "674.fits.gz", "350.fits.gz", "README.pdf"]
+SPIRE_ENDINGS = ["898.xml", "898.jpg", "141.fits.gz", "045.fits.gz", "952.fits.gz",
+                 "974.fits.gz", "715.fits.gz", "547.fits.gz", "770.fits.gz",
+                 "856.fits.gz", "148.fits.gz", "025.fits.gz", "538.fits.gz",
+                 "070.fits.gz", "434.fits.gz", "637.fits.gz", "835.fits.gz",
+                 "372.fits.gz", "248.fits.gz", "README.pdf"]
 
-    def access_archive_with_retries(self, f, params):
-        for _ in range(self.retries):
-            try:
-                res = f(**params)
-                return res
-            except ChunkedEncodingError:
-                pass
-        return None
 
-    def test_download_data_observation_pacs(self, tmp_path):
-        obs_id = "1342191813"
-        parameters = {'retrieval_type': "OBSERVATION",
-                      'observation_id': obs_id,
-                      'instrument_name': "PACS",
-                      'product_level': 'LEVEL3',
-                      'cache': False,
-                      'download_dir': tmp_path}
-        expected_res = Path(tmp_path, obs_id + ".tar")
-        hsa = HSAClass()
-        res = self.access_archive_with_retries(hsa.download_data, parameters)
-        if res is None:
-            pytest.xfail(f"Archive broke the connection {self.retries} times, unable to test")
-        assert Path(res) == expected_res
-        assert Path(res).is_file()
-        with tarfile.open(res) as tar:
-            chksum = [m.chksum for m in tar.getmembers()]
-        assert chksum.sort() == pacs_chksum.sort()
+@pytest.mark.parametrize(
+    "method,kwargs,expected_filename,expected_endings",
+    [
+        ("download_data", {}, "1342191813.tar", PACS_ENDINGS),
+        ("download_data", {"filename": "output_file"}, "output_file.tar", PACS_ENDINGS),
+        ("download_data", {"compress": "true"}, "1342191813.tgz", PACS_ENDINGS),
+        (
+            "download_data",
+            {
+                "observation_id": "1342191188",
+                "instrument_name": "SPIRE",
+                "product_level": "LEVEL2",
+            },
+            "1342191188.tar",
+            SPIRE_ENDINGS,
+        ),
+        ("get_observation", {}, "1342191813.tar", PACS_ENDINGS),
+    ],
+)
```

Comment:
this is madness, empty lines for no good reason, yet logically same lines are having different indents just because black. 

Please do use longer lines, manual edit, or anything that's not black to have consistency. And below, too, we do have line length up to 120, use that instead of having 3 separate lines

Reply:
There's fewer lines of code now.

Reply:
Thanks! If you know the right way to turn tools like black off on the config level, please open a PR to the config files, so I won't need to bikeshed again.

Reply:
I think the best option would be to set up an automatic code formatter that enforces a code style that is different from `black`, but I'm not too familiar with the alternatives. Perhaps [YAPF](https://pypi.org/project/yapf/) can be configured to achieve the desired result.